keel-api/src/main/kotlin/com/netflix/spinnaker/keel/api/plugins/ConstraintEvaluator.kt

```diff
@@ -60,18 +61,33 @@ interface ConstraintEvaluator<T : Constraint> : SpinnakerExtensionPoint {
   val eventPublisher: EventPublisher
 
   /**
-   * returns true if a constraint should be run for every environment in every delivery config, without being
+   * @return true if a constraint should be run for every environment in every delivery config, without being
    * exposed to delivery config author.
    */
   fun isImplicit(): Boolean = false
 
   /**
-   * TODO: Docs
+   * @return true if the constraint passes, false otherwise.
    */
   fun canPromote(
     artifact: DeliveryArtifact,
     version: String,
     deliveryConfig: DeliveryConfig,
     targetEnvironment: Environment
   ): Boolean
+
+  /**
+   * @return a constraint state object that captures the current state of the constraint.
+   * This will be used for stateless constraints to construct and save a summary of the state
```

Comment:
Since this only applies to stateless constraints, would it make sense to create a new `StatelessConstraintEvaluator` base class to host this function, and refactor the implementations to extend it?

Reply:
Hmm ya, it might. Let me test that out.

Reply:
This was sooooo much better, great suggestion @luispollo!